src/YarpUtilities/src/TransformHelper.cpp

```diff
@@ -0,0 +1,124 @@
+// SPDX-FileCopyrightText: Fondazione Istituto Italiano di Tecnologia (IIT)
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <yarp/os/LogStream.h>
+
+#include <WalkingControllers/YarpUtilities/TransformHelper.h>
+
+void WalkingControllers::YarpUtilities::TransformHelper::convertTransform(const iDynTree::Transform& transform, yarp::sig::Matrix& buffer)
+{
+    buffer.resize(4, 4);
+    buffer.zero();
+    iDynTree::Matrix4x4 homTrans = transform.asHomogeneousTransform();
+
+    // Here we explot the fact that both YARP and iDynTree store
+    // Matrices in row major form.
+    memcpy(buffer.data(), homTrans.data(), 4 * 4 * sizeof(double));
+}
+
+WalkingControllers::YarpUtilities::TransformHelper::~TransformHelper()
+{
+    close();
+}
+
+bool WalkingControllers::YarpUtilities::TransformHelper::configure(const yarp::os::Searchable& config)
+{
+    //From the GENERAL options
+    double period = config.check("sampling_time", yarp::os::Value(0.01)).asFloat64();
+    std::string prefix = "/" + config.check("name", yarp::os::Value("walking-coordinator")).asString();
+
+
+    m_rootFrame = config.check("rootFrame", yarp::os::Value("world")).asString();
+    m_baseFrameName = config.check("baseFrame", yarp::os::Value("root_link")).asString();
+    m_joystickFrameName = config.check("joystickFrame", yarp::os::Value("joystick")).asString();
+
+    if (config.check("additionalFrames"))
+    {
+        yarp::os::Value additionalFrames = config.find("additionalFrames");
+        if (!additionalFrames.isList())
+        {
+            yError() << "[TransformHelper::initialize] The additionalFrames is present, but it is not a list. No additional frame will be published.";
+        }
+        else
+        {
+            yarp::os::Bottle* frames = additionalFrames.asList();
+            for (int i = 0; i < frames->size(); ++i)
+            {
+                m_frames.push_back(frames->get(i).asString());
+            }
+        }
+    }
+
+    if (config.check("runServer", yarp::os::Value(false)).asBool())
+    {
+        yarp::os::Bottle& conf = config.findGroup("SERVER");
+
+        if (conf.isNull())
+        {
+            yError() << "[TransformHelper::initialize] Unable to get the group SERVER from the configuration file.";
+            return false;
+        }
+        if (!m_serverDriver.open(conf))
+        {
+            yError() << "[TransformHelper::initialize] Unable to open the server device with the following options:" << conf.toString();
+            return false;
+        }
+    }
+
+    yarp::os::Property tfClientCfg;
+    tfClientCfg.put("device", config.check("device", yarp::os::Value("frameTransformClient")).asString());
+    tfClientCfg.put("filexml_option", config.check("filexml_option", yarp::os::Value("ftc_yarp_only.xml")).asString());
```

Comment:
what is `ftc_yarp_only.xml`?

Reply:
It is the file defined internally by ``yarp`` to specify the configuration of the client. See https://www.yarp.it/latest/group__Fts__ftc__config.html

The file is stored here: https://github.com/robotology/yarp/tree/e52d732739b15c1ffcd14b29ca7bc42987b56931/src/devices/frameTransformClient/robotinterface_xml